app/command_channel_header.go

```diff
@@ -62,8 +62,8 @@ func (me *HeaderProvider) DoCommand(a *App, args *model.CommandArgs, message str
 
 	case model.CHANNEL_GROUP, model.CHANNEL_DIRECT:
 		// Modifying the header is not linked to any specific permission for group/dm channels, so just check for membership.
-		channelMember, err := a.GetChannelMember(args.ChannelId, args.Session.UserId)
-		if err != nil || channelMember == nil {
+		channelMember, channelErr := a.GetChannelMember(args.ChannelId, args.Session.UserId)
```

Comment:
This seems like a good example of the still experimental `go vet -shadow` going too far: forcing us to declare a different variable name just to avoid shadowing the `err` declared on line 38. I think this decreases readability.

Personally, I think we should allow explicitly shadowing within conditional declarations, e.g.:
```
if channelMember, err := a.GetChannelMember(args.ChannelId, args.Session.UserId); channelErr != nil || channelMember == nil {
    return  &model.CommandResponse{/* ... */}
}
```

When you can't use the inline form, you're in a position to want the variables outside the scope of the conditional block, and you're unlikely to worry about shadowing in that case.

Reply:
Yeah, I think pre-declaring `channelMember` here is probably the cleanest option.